scheduling/storage_provider.go

```diff
@@ -52,6 +56,56 @@ func (m *MemoryStorageProvider) Get(key string, prefix bool) ([]string, error) {
 	return result, nil
 }
 
+type ETCDStorageProvider struct {
+	client *clientv3.Client
+	ctx    context.Context
+}
+
+func NewETCDStorageProvider(client *clientv3.Client, ctx context.Context) *ETCDStorageProvider {
+	return &ETCDStorageProvider{client: client, ctx: ctx}
+}
+
+func (etcd *ETCDStorageProvider) Set(key string, value string) error {
+	if key == "" {
+		return fmt.Errorf("key cannot be empty")
+	}
+	if value == "" {
+		return fmt.Errorf("value cannot be empty")
```

Comment:
small suggestion. might be useful to know which key was trying to set the blank value (the key should be something after passing the first check)  "value cannot be empty for key: <key>" or something.

Reply:
so let's say that `set` is somehow invoked with ("myKey", "") 
the first if block will pass since `key` has a value, then the 2nd if block will return the error "value cannot be empty" but with no mention of which key. but if the logs already reference which key that error refers to then resolve this convo at will!

Reply:
Makes sense, will fix that